### PR TITLE
tpm2: Implement TPM2_TestParms

### PR DESCRIPTION
**Pull request by Foxboron**

### CODE REVIEW

File: tpm2/tpm2.go [hunk 1]
```diff
@@ -1596,6 +1596,28 @@ type GetCapabilityResponse struct {
 	CapabilityData TPMSCapabilityData
 }
 
+// TestParms is the input to TPM2_TestParms.
+// See definition in Part 3, Commands, section 30.3
+type TestParms struct {
+	// Algorithms paramteres to be validates
+	Paramteres TPMTPublicParms
```
Comment (chrisfenner) [inline tpm2/tpm2.go:1603]: nit: Parameters

File: tpm2/test/test_parms_test.go [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+					},
+				),
+			},
+			true,
+		},
+	} {
+		grc := TestParms{Paramteres: tt.parms}
+		_, err := grc.Execute(thetpm)
+
+		if err == nil && tt.shouldfail {
```
Comment (chrisfenner) [inline tpm2/test/test_parms_test.go:104]: Optional: you could replace `shouldFail` with a `wantErr error` in the anonymous test struct, then use [`errors.Is`](https://pkg.go.dev/errors#example-Is) for this check. This allows you to assert particular TPM error codes (e.g., `TPM_RC_VALUE`)

**Review by chrisfenner — CHANGES_REQUESTED** [review]
Sorry for the delayed response.

Looks good, just needs a typo fix. I made one other optional suggestion if you're up for it.

File: tpm2/test/test_parms_test.go (hunk 2 above)
Comment (Foxboron) [reply tpm2/test/test_parms_test.go]: I made an attempt :)

**Comment by chrisfenner** [comment]
(not sure if the force-push worked as intended; the linter that was updated in #353 now notices the typo)

**Comment by Foxboron** [comment]
> (not sure if the force-push worked as intended; the linter that was updated in https://github.com/google/go-tpm/pull/353 now notices the typo)

Forgot to pull :upside_down_face:

**Comment by Foxboron** [comment]
And it turns out i misspelled the word *twice* fixing it up. Great job.

**Comment by chrisfenner** [comment]
Perfect, looks right now. Thank you for taking me up on the optional testing suggestion! After linters and tests finish up I'll merge.

**Review by chrisfenner — APPROVED** [review]

**Comment by chrisfenner** [comment]
> And it turns out i misspelled the word _twice_ fixing it up. Great job.

Good to know I'm not the only person who does this.

**Pull request merged**